app/src/main/java/com/yacgroup/yacguide/LaunchActivity.kt

```diff
@@ -84,19 +86,92 @@ class LaunchActivity : AppCompatActivity() {
             true
         } else {
             false
         }
     }
+
+    /**
+     * Check whether the settings need to be migrated or not.
+     */
+    private fun _checkForSettingsMigration() {
+        if (_customSettings.getInt(getString(R.string.pref_key_pref_version), 1) == 1) {
+            _migrateSettings()
+        }
+    }
+
+    /**
+     * Migrate settings, if necessary, to fix issue https://github.com/YacGroup/yacguide/issues/453.
+     */
+    private fun _migrateSettings() {
+        val settingsResourceMap = mapOf(
+            R.string.order_tourbook_chronologically to
+                    Pair(R.string.pref_key_order_tourbook_chronologically, R.bool.pref_default_order_tourbook_chronologically),
+            R.string.only_official_summits to
+                    Pair(R.string.pref_key_only_official_summits, R.bool.pref_default_only_official_summits),
+            R.string.only_official_routes to
+                    Pair(R.string.pref_key_only_official_routes, R.bool.pref_default_only_official_routes),
+            R.string.count_summits to
+                Pair(R.string.pref_key_count_summits, R.bool.pref_default_count_summits),
+            R.string.count_massifs to
+                Pair(R.string.pref_key_count_massifs, R.bool.pref_default_count_massifs),
+            R.string.count_boulders to
+                    Pair(R.string.pref_key_count_boulders, R.bool.pref_default_count_boulders),
+            R.string.count_caves to
+                    Pair(R.string.pref_key_count_caves, R.bool.pref_default_count_caves),
+            R.string.count_unofficial_rocks to
+                    Pair(R.string.pref_key_count_unofficial_rocks, R.bool.pref_default_count_unofficial_rocks),
+            R.string.count_prohibited_rocks to
+                    Pair(R.string.pref_key_count_prohibited_rocks, R.bool.pref_default_count_prohibited_rocks),
+            R.string.count_collapsed_rocks to
+                    Pair(R.string.pref_key_count_collapsed_rocks, R.bool.pref_default_count_collapsed_rocks),
+            R.string.count_only_leads to
+                    Pair(R.string.pref_key_count_only_leads, R.bool.pref_default_count_only_leads),
+            R.string.colorize_tourbook_entries to
+                    Pair(R.string.pref_key_colorize_tourbook_entries, R.bool.pref_default_colorize_tourbook_entries),
+            R.string.lead to
+                    Pair(R.string.pref_key_color_lead, R.color.pref_default_color_lead),
+            R.string.follow to
+                    Pair(R.string.pref_key_color_follow, R.color.pref_default_color_follow)
+        )
+        _customSettings.let {
+            it.edit().apply {
+                for ((oldResource, newResourcePair) in settingsResourceMap) {
+                    val oldKey = resources.getString(oldResource)
+                    val newKey = resources.getString(newResourcePair.first)
+                    Log.d("[PREF]", "Checking for deprecated key '$oldKey'.")
```

Comment:
I think, if we want to introduce logging, we should check for creating a real logger module or so. So I'm not sure if we should really introduce it as part of this PR.

Reply:
OK, I agree. It was mainly for debugging but I will remove it.